Access log interceptor (#1327)
vivid-planet/comet: .changeset/olive-comics-knock.md

---
"@comet/cms-api": minor
"comet-docs": minor
---

> Add AccessLog Interceptor where request data is logged in the standard console. The access log allows to see who accessed what data and when. This information for example can be used to identify security vulnerabilities.

Reviewer: ```suggestion
Add AccessLog Interceptor to log information about the request to standard output. The informations contains information about the requester and the request itself. This can be useful for fulfilling legal requirements regarding data integrity or for forensics.
```

Reviewer 2: ```suggestion
Add access logging to log information about the request to standard output. The log contains information about the requester and the request itself. This can be useful for fulfilling legal requirements regarding data integrity or for forensics.
```

The usage of an interceptor is an implementation detail IMO

> There are two options for implementing the module in the app:

Reviewer 2: ```suggestion
There are two ways to integrate logging into an application:
```

> **1 option:**

Reviewer 2: ```suggestion
**First option: Use the default implementation**
```

```ts
imports: [
    ...
    AccessLogModule,
    ...
]
```

> **2 option:**

Reviewer 2: ```suggestion
**Second option: Configure logging**
```

> With the second option, there is the opportunity to add a callback function called `filterRequest`. This function provides the ability to disable logging for various functionalities.

Reviewer 2: ```suggestion
Use the `filterRequest` to prevent logging for specific requests. For instance, one may filter requests for system users.
```

---
"@comet/cms-api": minor
"comet-docs": minor
---

> Add AccessLog Interceptor

Reviewer 2: Please be a little more specific: What is it, why do we add it, how can one integrate it into an application
